File: Java/Data Structures & Algorithms Module - Java/Week10/BSTAttempt5/BSTApp.java
/*
BSTApp.java
Billy Cussen
09/12/2020
*/

public class BSTApp<T>{

    public static void main(String [] args){

        BST<Integer> myBST = new BST<Integer>();

        try{
            myBST.inOrder();
        }catch(NullPointerException e){
            System.out.println("EXCEPTION: "+e.getMessage());
        }
        System.out.println("Empty: "+myBST.isEmpty());
        System.out.println("Size: "+myBST.size());
        System.out.println("Depth: "+myBST.depth());
        System.out.println("Contains 20: "+myBST.contains(20));
        System.out.println("Contains 2000: "+myBST.contains(2000));
        try{
            System.out.println("Max: "+myBST.max());
        }catch(NullPointerException e){
            System.out.println("EXCEPTION: "+e.getMessage());
        }
        try{
            System.out.println("Min: "+myBST.min());
        }catch(NullPointerException e){
            System.out.println("EXCEPTION: "+e.getMessage());
        }

        myBST.insert(20);
        myBST.insert(15);
        myBST.insert(10);
        myBST.insert(5);
        myBST.insert(7);
        myBST.insert(25);
        myBST.insert(22);

        myBST.inOrder();
        System.out.println("Empty: "+myBST.isEmpty());
        System.out.println("Size: "+myBST.size());
        System.out.println("Max: "+myBST.max());
        System.out.println("Min: "+myBST.min());
        System.out.println("Depth: "+myBST.depth());
        System.out.println("Contains 20: "+myBST.contains(20));
        System.out.println("Contains 2000: "+myBST.contains(2000));

    }

}

File: Python/Data Structures & Algorithms Module - Python/Week10/BST Attempt 4/BSTApp.py
"""
BSTApp.py
Billy Cussen
10/12/2020
"""

from BST import BST

myBST = BST()

print("Empty: "+str(myBST.isEmpty()))
print("Size: "+str(myBST.size()))
print("Max: "+str(myBST.max()))
print("Min: "+str(myBST.min()))
myBST.inOrder()
print("\n")

myBST.insert(20)
myBST.insert(15)
myBST.insert(17)
myBST.insert(12)
myBST.insert(30)
myBST.insert(22)

print("Empty: "+str(myBST.isEmpty()))
print("Size: "+str(myBST.size()))
print("Max: "+str(myBST.max()))
print("Min: "+str(myBST.min()))
myBST.inOrder()
print("\n")


print("Remove 22: "+str(myBST.remove(22)))
print("Remove 30: "+str(myBST.remove(30)))

print("Empty: "+str(myBST.isEmpty()))
print("Size: "+str(myBST.size()))
print("Max: "+str(myBST.max()))
print("Min: "+str(myBST.min()))
myBST.inOrder()
print("\n")